dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatHistory.cs

```diff
@@ -69,34 +79,76 @@ public ChatHistory(IEnumerable<ChatMessageContent> messages)
     public void AddMessage(AuthorRole authorRole, ChatMessageContentItemCollection contentItems, Encoding? encoding = null, IReadOnlyDictionary<string, object?>? metadata = null) =>
         this.Add(new ChatMessageContent(authorRole, contentItems, null, null, encoding, metadata));
 
+    /// <summary>
+    /// <param name="name">Name of the author of the message</param>
+    /// <param name="authorRole">Role of the message author</param>
+    /// <param name="contentItems">Instance of <see cref="ChatMessageContentItemCollection"/> with content items</param>
+    /// <param name="encoding">Encoding of the message content</param>
+    /// <param name="metadata">Dictionary for any additional metadata</param>
+    /// </summary>
+    public void AddMessage(string? name, AuthorRole authorRole, ChatMessageContentItemCollection contentItems, Encoding? encoding = null, IReadOnlyDictionary<string, object?>? metadata = null) =>
+        this.Add(new ChatMessageContent(name, authorRole, contentItems, null, null, encoding, metadata));
+
     /// <summary>
     /// Add a user message to the chat history
     /// </summary>
     /// <param name="content">Message content</param>
     public void AddUserMessage(string content) =>
         this.AddMessage(AuthorRole.User, content);
 
+    /// <summary>
+    /// Add a user message to the chat history
+    /// </summary>
+    /// <param name="name">Name of the author of the message</param>
+    /// <param name="content">Message content</param>
+    public void AddUserMessageWithName(string? name, string content) =>
```

Comment:
This pattern doesn't scale i.e. what happens the next time a property is added.

Let's leave this out as it will be very difficult (if not impossible) to remove this in future.

Reply:
Try to found where is it